### PR TITLE
Convert to typescript

### PR DESCRIPTION
**Pull request by rictic**

### CODE REVIEW

**Comment by rictic** [comment]
This should help with #19

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 language: node_js
 sudo: false
 node_js:
+  - "6"
   - "5"
```
Comment (justinfagnani) [inline .travis.yml]: I'm ready to pull testing support for 5 and 0.12

File: dom5.ts [hunk 2]
```diff
@@ -1,24 +1,49 @@
...
- * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
  */
 
 // jshint node: true
 'use strict';
 
-var cloneObject = require('clone');
+declare module 'parse5' {
```
Comment (justinfagnani) [inline dom5.ts]: why are we declaring the parse5 module in dom5.ts?

**Comment by justinfagnani** [comment]
LGTM, w/ one question about the parse5 module

File: .travis.yml (hunk 1 above)
Comment (rictic) [inline .travis.yml]: Ack. IMO we should drop support with a breaking change.

File: dom5.ts (hunk 2 above)
Comment (rictic) [inline dom5.ts]: Good point, worked out how to move to custom_typings dir

**Comment by inikulin** [comment]
Hi guys,

You can use [typedoc](https://github.com/TypeStrong/typedoc) to automatically generate docs from typescript. It could resolve #19 with a little effort. I've recently used it for [parse5 docs](http://inikulin.github.io/parse5/).

**Comment by rictic** [comment]
Nice! Those docs look pretty sweet

**Pull request merged**